### PR TITLE
[Iga] Penalty Coupling Condition

### PR DESCRIPTION
**Pull request by tteschemacher**
Hey,

I would like to add the condition for penalty coupling to the Iga Application. This is to couple only G0 continuity, G1 will come in later PRs.

This PR requires a merge of the coupling geometries, first #5390 . However, it shows how those geometries can be used.

### CODE REVIEW

**Comment by adityaghantasala** [comment]
Hey 
Is t possible to generalize this and make it available for other applications too ? ( From the core ?)

**Comment by tteschemacher** [comment]
I think that would be quite nice. I was already thinking about a condition which gets a list of DOFs for which the continuity should be enforced (DISPS, ROT, ...) - but here I didn't know how to pass this, maybe somebody has an idea? This would then remove the dependency of the IgaFlags and would be completely independent, also on the used Dofs. From a mathematical point of view this condition is already not dependent on Iga.

**Comment by oberbichler** [comment]
@adityaghantasala I am working on a generalized coupling implementation and have implemented an coupling element for displacement and rotations which is completely independent from the element type and the geometric description that can switch between coupling techniques (penalty/(augmented) lagrange). At the moment I am testing it just with IGA. What would be your use case?

File: applications/IgaApplication/iga_application_variables.cpp [hunk 1]
```diff
@@ -33,4 +33,7 @@ KRATOS_CREATE_3D_VARIABLE_WITH_COMPONENTS(POINT_LOAD)
 KRATOS_CREATE_3D_VARIABLE_WITH_COMPONENTS(LINE_LOAD)
 KRATOS_CREATE_3D_VARIABLE_WITH_COMPONENTS(SURFACE_LOAD)
 
+//Penalty Variables
+KRATOS_CREATE_VARIABLE(double, PENALTY_FACTOR)
```
Comment (loumalouomega) [inline applications/IgaApplication/iga_application_variables.cpp:37]: There is already a PENALTY parameters in the core, called INITIAL_PENALTY

**Review by loumalouomega — COMMENTED** [review]
Just some comment

File: applications/IgaApplication/iga_application_variables.cpp (hunk 1 above)
Comment (tteschemacher) [reply applications/IgaApplication/iga_application_variables.cpp:37]: Thanks for the hint! INITIAL_PENALTY is wuite missleading don't you think so?

**Comment by loumalouomega** [comment]
Yep, but it is the one I use. It is called that way because in the explicit
schemes , and sometimes in the implicit, the penalty is adapted. So it
represent the initial value before adapt it.

**Review by RiccardoRossi — DISMISSED** [review]
ok

**Review by andreas-apostolatos — APPROVED** [review]
Hey Tobi,

thanks a lot for the development, I am very happy with it :)

Greetings!

**Pull request merged**